host/volume/api/http.go

```diff
@@ -0,0 +1,127 @@
+package volumeapi
+
+import (
+	"encoding/json"
+	"fmt"
+	"net/http"
+
+	"github.com/flynn/flynn/Godeps/_workspace/src/github.com/julienschmidt/httprouter"
+	"github.com/flynn/flynn/host/volume"
+	"github.com/flynn/flynn/host/volume/zfs"
+	"github.com/flynn/flynn/pkg/httphelper"
+	"github.com/flynn/flynn/pkg/random"
+)
+
+type HTTPAPI struct {
+	vman *volume.Manager
+}
+
+func NewHTTPAPI(vman *volume.Manager) *HTTPAPI {
+	return &HTTPAPI{vman: vman}
+}
+
+func RegisterRoutes(api *HTTPAPI, r *httprouter.Router) {
```

Comment:
I'd make this a `func (api *HTTPAPI) RegisterRoutes(r *httprouter.Router)`, like in Sampi.